libs/agno/agno/knowledge/knowledge.py

```diff
@@ -957,11 +967,18 @@ def add_reader(self, reader: Reader):
         return reader
 
     def get_readers(self) -> List[Reader]:
-        """Get all available readers."""
+        """Get all currently loaded readers (only returns readers that have been used)."""
         if self.readers is None:
-            return []
+            self.readers = {}
+        
         return list(self.readers.values())
 
+    def get_loaded_reader_types(self) -> List[str]:
+        """Get list of currently loaded reader types."""
+        if self.readers is None:
+            return []
+        return list(self.readers.keys())
```

Comment:
Are both of these needed? You can just do `get_readers` and either access the keys of values depending on your usecase